packages/logger/src/logger.injectable.ts

```diff
@@ -2,28 +2,116 @@
  * Copyright (c) OpenLens Authors. All rights reserved.
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
-import { getInjectable, getInjectionToken } from "@ogre-tools/injectable";
+import { kebabCase, toUpper } from "lodash/fp";
+import {
+  DiContainerForInjection,
+  getInjectable,
+  getInjectionToken,
+  lifecycleEnum,
+} from "@ogre-tools/injectable";
 import type { Logger } from "./logger";
 import { winstonLoggerInjectable } from "./winston-logger.injectable";
+import { pipeline } from "@ogre-tools/fp";
 
+/** @deprecated Use specific injectionToken, eg. logErrorInjectionToken */
 export const loggerInjectionToken = getInjectionToken<Logger>({
   id: "logger-injection-token",
 });
 
 export const loggerInjectable = getInjectable({
   id: "logger",
-  instantiate: (di): Logger => {
-    const baseLogger = di.inject(winstonLoggerInjectable);
-
-    return {
-      debug: (message, ...data) => baseLogger.debug(message, ...data),
-      info: (message, ...data) => baseLogger.info(message, ...data),
-      warn: (message, ...data) => baseLogger.warn(message, ...data),
-      error: (message, ...data) => baseLogger.error(message, ...data),
-      silly: (message, ...data) => baseLogger.silly(message, ...data),
-    };
-  },
+  instantiate: (di): Logger => ({
+    debug: di.inject(logDebugInjectionToken),
+    info: di.inject(logInfoInjectionToken),
+    warn: di.inject(logWarningInjectionToken),
+    error: di.inject(logErrorInjectionToken),
+    silly: di.inject(logSillyInjectionToken),
+  }),
 
   decorable: false,
   injectionToken: loggerInjectionToken,
 });
+
+export type LogFunction = (message: string, ...data: any[]) => void;
+
+export const logDebugInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-debug-injection-token",
+});
+
+export const logInfoInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-info-injection-token",
+});
+
+export const logWarningInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-warning-injection-token",
+});
+
+export const logErrorInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-error-injection-token",
+});
+
+export const logSillyInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-silly-injection-token",
+});
+
+const screamingKebabCase = (str: string) => pipeline(str, kebabCase, toUpper);
+
+const getLogFunctionFor =
+  (scenario: keyof Logger) =>
+  (di: DiContainerForInjection): LogFunction => {
+    const winstonLogger = di.inject(winstonLoggerInjectable);
+
+    return (message, ...data) => {
+      winstonLogger[scenario](
+        di.sourceNamespace
+          ? `[${screamingKebabCase(di.sourceNamespace)}]: ${message}`
```

Comment:
Notice: `di.sourceNamespace` is a new thing in `injectable`, making this auto-prefixing possible.

Reply:
why not use the already existing `prefixedLogger`?

Reply:
Because after this you don't have to worry about prefixing at all anymore. It's all done automatically within `loggerFeature`. Usage doesn't have to know about it.

Reply:
Because of inconsistency, boilerplate and unnecessary complexity.

Reply:
More interestingly, I'm thinking how `di.sourceNamespace` could be leveraged elsewhere. With it, we can have stuff that is guaranteed to have a Feature-specific instance.

1. We can simplify injections eg. for less required `instantiationParameters`. 
1. We can have error handling that can identify problematic Features.
1. We can have profiling that can identify slow Features.
1. We can have Features able to only access its own directory in file-system.
1. We can solve many naming collision issues, as namespace is always guaranteed unique.
1. ...best ideas win a prize at next team event ;)

Reply:
profile the loading time of the app so we can get closer to sub 2second loading